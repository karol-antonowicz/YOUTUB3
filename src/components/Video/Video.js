import React from 'react';
import './Video.scss';

const BASE_EMBED_URL = 'https://www.youtube.com/embed/';

export const Video = props => {
    if (!props.id) {
        return null;
    }
    const embedUrl = `${BASE_EMBED_URL}${props.id}`;

    return (
        <iframe 
        width="560" height="315" 
        src={embedUrl}
        frameborder="0" 
        allow="accelerometer; 
        autoplay; 
        encrypted-media; 
        gyroscope; 
        picture-in-picture" 
        allowfullscreen>
        </iframe>
    )
}