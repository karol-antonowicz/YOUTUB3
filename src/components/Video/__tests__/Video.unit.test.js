import React from 'react';
import {shallow} from 'enzyme';
import {Video} from '../Video';

describe('Video', ()=> {
    test('renders Video component with some id', ()=> {
        const wrapper = shallow(<Video id='56sd65ds'/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('nothing rendered as no id from props given', ()=> {
        const wrapper = shallow(<Video/>);
        expect(wrapper).toMatchSnapshot();
    })
});