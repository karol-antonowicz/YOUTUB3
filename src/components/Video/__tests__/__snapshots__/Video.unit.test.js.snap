// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video nothing rendered as no id from props given 1`] = `""`;

exports[`Video renders Video component with some id 1`] = `
<div
  className="video-container"
>
  <div
    className="video"
  >
    <iframe
      allow="accelerometer;  autoplay;  encrypted-media;  gyroscope;  picture-in-picture"
      allowfullscreen={true}
      className="video-player"
      frameborder="0"
      src="https://www.youtube.com/embed/56sd65ds?autoplay=1"
      title="video"
    />
  </div>
</div>
`;
